File: 2021/May/Day 17/addDigits.py
# Source :https://leetcode.com/problems/add-digits/submissions/
class Solution:
    def addDigits(self, num: int) -> int:
        digits =0;
        #print("Number is : ", num)
        if(int(num/10)==0):
            return num;
        while(num):
            digits+= int(num%10);
            num = int(num/10);
        #print(digits)
        return self.addDigits(digits);

File: 2022/November/Day 29/addDigits.cpp
// Source : https://leetcode.com/problems/add-digits/submissions/
class Solution {
public:
    int addDigits(int num) {
        int t=num;
        int debug=0;
        while(t>=10)
        {
            t=0;
        while(num)
        {
            t+=num%10;
            num/=10;
        }

            if(debug)
            cout<<"Num is "<<num<<" T : "<<t<<endl;
            num=t;
        }

        return t;
    }
};
